### PR TITLE
<DOCS>V4 virtual world defining files

### PR DESCRIPTION
**Pull request by SEAJang**
# Pull Request Key Information

## Reason for change

There was no documentation related to creating infrastructure files

## What was changed

Added more documentation

## Intended Purpose

Provide users with a detailed description of how to set the different infrastructure files

## Testing Completed

No code testing was done as only the user manual was updated.

The hyperlinks that have existing entries were tested to function correctly. However, it is important to note that there are hyperlinks that are currently placeholders for the entries that do not exist in the documentation yet.

### CODE REVIEW

File: USER_MANUAL.md [hunk 1]
```diff
@@ -634,7 +652,310 @@ Content for the "Method Inputs" section goes here.
...
+| -1.79 | 10 |
+| 2.17 | 4 |
+| | 1.2|
+| | 2.2|
+
+--------------------------------------------------------------------------------
+
+#### Header
+
+**Description:** A user define row name, that corresponds to the [repairable emission rate source](#emission_rate_source-repairable)(or [non-repairable](#emission_rate_source-non-repairable)) references used in the virtual world parameters or in the infrastructure files.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:898]: Spelling/Grammar: defined not define

File: USER_MANUAL.md [hunk 2]
```diff
@@ -634,7 +652,310 @@ Content for the "Method Inputs" section goes here.
...
+**Notes on acquisition:** User defined
+
+### Data Use
+
+**Description:** Describes way the numerical information in the column is utilized. Valid inputs are:
+
+- `sample`: Sample random values as emission rates from the list provided
+- `dist`: Generate random values to use as emissions based on the distribution shape and scale provided
+
+**Notes on acquisition:** User defined. In the future, the ``fit`` functionality will be reimplemented to enable fitting of sample data to the most appropriate statistical curve.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:909]: Lets leave out the part saying we will reimplement fit in the future

File: USER_MANUAL.md [hunk 3]
```diff
@@ -634,7 +652,310 @@ Content for the "Method Inputs" section goes here.
...
+- month
+- year
+
+**Notes on acquisition:** User defined
+
+**Notes of caution:** All units are converted into base grams per second. Conversions are based on standard atmospheric temperature and pressure conditions and may have slight variations.
+
+#### Source (Emission file)
+
+**Description:** Each row defines a single input. Depending on the [data use](#data-use) input, the values specify either individual emission rates (`sample`),  or the shape and scale of the distribution (`dist`).
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:956]: Nit: Can you provide a little more detail about this

**Review by ThomasGalesloot — CHANGES_REQUESTED** [review]
Great work! See my comments for some description nitpicks. Some general feedback: For each file I feel it would be helpful for users to describe what a row is supposed to represent. Also the formatting felt inconsistent when reading the documentation - can we tweak it a bit so that it feels like consistently structured

File: USER_MANUAL.md [hunk 4]
```diff
@@ -634,7 +652,310 @@ Content for the "Method Inputs" section goes here.
...
+- {method}_[deploy_year](#deployment_year)
+- {method}_[deploy_month](#deployment_month)
+- {method}_[spatial](#spatial)
+- {method}_[survey_time](#survey_time)
+- {method}_[survey_cost](#per_site-cost)
+- [{method}_site_deployment](#method_site_deployment)
+
+**Note:** As with most name-related parameters in LDAR-Sim, the values in the `site_type` and `equipment` columns are case-sensitive and must be consistent across all related files. Any method specific parameters also require the same care with the case-sensitivity and consistency in all relevant files with the method name.
+
+See [sites file](#sites-file) for details on how to set the parameter for simulation.
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply USER_MANUAL.md:702]: This is to link to the virtual world parameters.

File: USER_MANUAL.md [hunk 5]
```diff
@@ -634,7 +652,310 @@ Content for the "Method Inputs" section goes here.
...
+- [repairable_multiple_emissions_per_source](#multiple_emissions_per_source-repairable)
+- [non_repairable_emissions_rate_source](#emissions_rate_source_non-repairable)
+- [non_repairable_emissions_production_rate](#emissions_production_rate-non-repairable)
+- [non_repairable_duration](#duration-non-repairable)
+- [non_repairable_multiple_emissions_per_source](#multiple_emissions_per_source-non-repairable)
+
+Method specific columns:
+
+- {method}_[surveys_per_year](#surveys_per_year)
+- {method}_[deploy_year](#deployment_year)
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply USER_MANUAL.md:693]: Double-checked the code, and we only added it for site types, which doesn't make much sense. Fixed to also be able to set this for sites.

File: USER_MANUAL.md [hunk 6]
```diff
@@ -634,7 +652,310 @@ Content for the "Method Inputs" section goes here.
...
+At a minimum it must contain the following columns:
+
+- [site_ID](#site_id)
+- [lat](#lat)
+- [lon](#lon)
+- [site_type](#site_type-sites-file)
+
+Other optional columns consist of the following:
+
+- [equipment](#equipment-sites-file)
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply USER_MANUAL.md:678]: That seems kind of redundant? This section is doing that.

File: USER_MANUAL.md [hunk 7]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+This parameter can also be set in the following files:
+
+- [Virtual World Parameter file](#emissions_rate_source-propagating-parameter)
+- [Site type File](#non_repairable_emissions_rate_source-site-type-file)
+- [Equipment File](#non_repairable_emissions_rate_source-equipment-file)
+- [Source File](#emissions_rate_source-source-file)
+
+#### non_repairable_emissions_production_rate (Sites file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See [emissions_production_rate](#emissions_production_rate-propagating-parameter) for more details on what this parameter represents. In this file, this is a single numeric value, representing the emission production rate per day for the non-repairable emissions at the given site.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1900]: same comment as emissions production rate above

File: USER_MANUAL.md [hunk 8]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+This parameter can also be set in the following files:
+
+- [Virtual World Parameter file](#emissions_production_rate-propagating-parameter)
+- [Site type File](#non_repairable_emissions_production_rate-site-type-file)
+- [Equipment File](#non_repairable_emissions_production_rate-equipment-file)
+- [Source File](#emissions_production_rate-source-file)
+
+#### non_repairable_duration (Sites file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See [duration](#duration-propagating-parameter) for more details on what this parameter represents. In this file, this is a single numeric value in days, representing the maximum duration of the non-repairable emissions at the given site.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1911]: same comment as for duration above

File: USER_MANUAL.md [hunk 9]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+This parameter can also be set in the following files:
+
+- [Virtual World Parameter file](#duration-propagating-parameter)
+- [Site type File](#non_repairable_duration-site-type-file)
+- [Equipment File](#non_repairable_duration-equipment-file)
+- [Source File](#duration-source-file)
+
+#### non_repairable_multiple_emissions_per_source (Sites file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See [multiple_emissions_per_source](#multiple_emissions_per_source-propagating-parameter) for more details on what this parameter represents. In this file, this is a `TRUE`/`FALSE` value, representing if there can be multiple emissions existing at a given time for non-repairable emissions for the given site.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1922]: same comment as for multiple emissions above

File: USER_MANUAL.md [hunk 10]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See[surveys_per_year](#surveys_per_year-propagating-parameter) for more details on what this parameter represents. In this file, this is a single integer representing the number of surveys to be conducted in a given year for the specified site for the specified method.
+
+This parameter can also be set in the following files:
+
+- [Method Parameter file](#surveys_per_year-propagating-parameter)
+- [Site type File](#method_surveys_per_year-site-type-file)
+
+#### {method}_deploy_years (Sites file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See [deploy_year](#deployment_years--propagating-parameter) for more details on what this parameter represents. In this file, this is a list of integers representing which years the specified method can be deployed on.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1942]: For the given site*

File: USER_MANUAL.md [hunk 11]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+This parameter can also be set in the following files:
+
+- [Method Parameter file](#deployment_years--propagating-parameter)
+- [Site type File](#method_deploy_years-site-type-file)
+
+#### {method}_deploy_months (Sites file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See[deploy_month](#deployment_months--propagating-parameter) for more details on what this parameter represents. In this file, this is a list of integers representing which months the specified method can be deployed on.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1951]: For the given site*

File: USER_MANUAL.md [hunk 12]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+This parameter can also be set in the following files:
+
+- [Method Parameter file](#deployment_months--propagating-parameter)
+- [Site type File](#method_deploy_months-site-type-file)
+
+#### {method}_spatial (Sites file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See[spatial](#spatial-propagating-parameter) for more details on what this parameter represents. In this file, this is a single value, representing the chance in which the given method can detect emissions at the given site.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1960]: I don't feel great about this one but I don't have a suggestion on how to improve

File: USER_MANUAL.md [hunk 13]
```diff
@@ -1549,58 +1729,912 @@ The following figure offers a visual guideline for the various parameters that c
...
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See [survey_cost](#per_site--propagating-parameter) for more details on what this parameter represents. In this file, this is a single value, representing the cost of the specified method to survey the given site.
+
+This parameter can also be set in the following files:
+
+- [Method Parameter file](#per_site--propagating-parameter)
+- [Site type file](#method_survey_cost-site-type-file)
+
+#### {method}_site_deployment (Sites file)
+
+**Description:** A True/False column that indicates whether the specified method will be deployed at the given site type. For example, if the value is set to `FALSE`, the emissions at the site will be simulated, but the method will not be deployed. This is similar to setting the[surveys per year](#surveys_per_year-propagating-parameter) to 0 for the site.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:1989]: Nit: 2 amendments.
1.  A True/False column that indicates whether the specified method will be deployed at the given site.
2 but the method will not be deployed at that site*

**Review by ThomasGalesloot — CHANGES_REQUESTED** [review]
Overall great work! Really loving the structure now. I've commented some changes for the sites file only, but can you also make the equivalent changes for the other infrastructure files as well. After this, I'd like to do one more review just checking for little errors and grammar mistakes, but then this should be ready.

File: USER_MANUAL.md [hunk 14]
```diff
@@ -126,10 +126,77 @@ Email: <sally@highwoodemissions.com>
       - [\<threshold\>](#threshold)
   - [10. Virtual World Defining Files](#10-virtual-world-defining-files)
     - [Sites File](#sites-file)
+      - [site\_ID](#site_id)
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply USER_MANUAL.md:129]: This will be a part of a different PR.

File: USER_MANUAL.md [hunk 15]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+|----|----|----|----|
+|group1|1|1|0|
+|group2|0|0|2|
+|group3|3|0|0|
+
+--------------------------------------------------------------------------------
+
+#### equipment (Equipment File)
+
+**Description:** A user-defined value that defines a specific equipment group for the simulation. These equipment groups influence how  `equipment level` [measurement scale](#measurement_scale)  methods detect and measure emissions.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2295]: note each row must be a unique equipment

File: USER_MANUAL.md [hunk 16]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+**Note:** It is case-sensitive and must remain consistent when provided in other infrastructure files, such as the [sites file](#sites-file) or the [site type file](#site-type-file)
+
+#### {component} (Equipment File)
+
+**Description:** The column header is a user defined value of a specific component in the simulation. The values in each column and row indicate how many counts of the given component(column) exists for the given equipment(row). These _equipment_ groups can serve as building blocks for constructing complex sites.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2301]: Nit: The column header is a user defined value corresponding the identifier of a specific component defining as part of the simulation.

File: USER_MANUAL.md [hunk 17]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+**Note of caution:** The `component` and `source` columns together act as a compound key. In simple terms, each component/source pair must be unique and exist only once.
+
+This column is case-sensitive and must match the corresponding columns provided in other infrastructure files.
+
+#### source (Source File)
+
+**Description:** A user-defined source represents a potential emission source for a given component. It is the smallest granular level considered for emission production.
+
+**Note:** The `source` column is often identical to the `emissions_rate_source` column. However it is in there incase users would like to use the same `emission_rate_sources`, for different `sources`.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2488]: Can we make it clear that the values in the source column are distinct from the emissions rate source column and you do need to populate both to use an emissions rate source

File: USER_MANUAL.md [hunk 18]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+**Description:** A True/False column that defines if the given emission source produces repairable(`TRUE`) or non-repairable(`FALSE`) emission.
+
+#### persistent (Source File)
+
+**Description:** A True/False column that defines if the given emission source produces persistent(`TRUE`) or non-persistent(`FALSE`) emission. Non-persistent emissions have a total active duration of the user set parameter [duration](#duration-propagating-parameter). However each active and inactive period is defined by the following parameters respectively, [active_duration](#active_duration-source-file) and [inactive_duration](#inactive_duration-source-file).
+
+#### active_duration (Source File)
+
+**Description:** The number of days that the given non-persistent emission is active for at a given time.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2502]: Instead of active in the description can we emitting? I think it's more clear

File: USER_MANUAL.md [hunk 19]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+#### inactive_duration (Source File)
+
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2505]: Instead of inactive, can we use non-emitting

File: USER_MANUAL.md [hunk 20]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+Non-Repairable Emissions:
+
+- [Virtual World Parameter file](#emissions_rate_source-propagating-parameter)
+- [Sites File](#non_repairable_emissions_rate_source-sites-file)
+- [Site type File](#non_repairable_emissions_rate_source-site-type-file)
+- [Equipment File](#non_repairable_emissions_rate_source-equipment-file)
+
+#### emissions_production_rate (Source file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See [emissions_production_rate](#emissions_production_rate-propagating-parameter) for more details on what this parameter represents. In this file, this is a single numeric value, representing the emission production rate of the given source at the given component per day.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2552]: This one didn't get updated to be in line with the new emissions rate source descriptions

File: USER_MANUAL.md [hunk 21]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+Non-Repairable emissions:
+
+- [Virtual World Parameter file](#multiple_emissions_per_source-propagating-parameter)
+- [Sites File](#non_repairable_multiple_emissions_per_source-sites-file)
+- [Site type File](#non_repairable_multiple_emissions_per_source-site-type-file)
+- [Equipment File](#non_repairable_multiple_emissions_per_source-equipment-file)
+
+#### {method}_spatial (Source file)
+
+**Description:** This is a propagating parameter that can be set at multiple levels of granularity. See[spatial](#spatial-propagating-parameter) for more details on what this parameter represents. In this file, this is a single value, representing the chance in which the given method can detect the possible emission source.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2612]: Not detect the source, but an emissions coming from that source.

File: USER_MANUAL.md [hunk 22]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+| -1.79 | 10 |
+| 2.17 | 4 |
+| | 1.2|
+| | 2.2|
+
+--------------------------------------------------------------------------------
+
+#### Header
+
+**Description:** A user defined row name, that corresponds to the [emissions rate source](#emissions_rate_source-propagating-parameter) references used in the virtual world parameters or in the infrastructure files.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2657]: column?

Comment (ThomasGalesloot) [reply USER_MANUAL.md:2657]: Can we say something along the lines of it's essential just an ID

File: USER_MANUAL.md [hunk 23]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+#### Data Use
+
+**Description:** Describes way the numerical information in the column is utilized. Valid inputs are:
+
+- `sample`: Sample random values as emission rates from the list provided
+- `dist`: Generate random values to use as emissions based on the distribution shape and scale provided
+
+#### Distribution Type
+
+**Description:** The name of a distribution from the scipy library, only relevant if the row above is set to `dist`.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2670]: instead of saying row above can we say data use

File: USER_MANUAL.md [hunk 24]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+**Notes on acquisition:** See [scipy documentation](https://docs.scipy.org/doc/scipy/reference/stats.html) for more details.
+
+#### Maximum Emission Rate
+
+**Description:** The maximum possible leak rate that can be sampled from a distribution, in grams per second.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2676]: emission rate not leak rate

File: USER_MANUAL.md [hunk 25]
```diff
@@ -1549,58 +1740,983 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+**Notes on acquisition:** User defined
+
+**Notes of caution:** All units are converted into base grams per second. Conversions are based on standard atmospheric temperature and pressure conditions and may have slight variations.
+
+#### Source (Emission file)
+
+**Description:** Each row defines a single input. Depending on the [data use](#data-use) input, the values specify either individual emission rates (`sample`) or the shape and scale of the distribution (`dist`).
+
+- sample: Each row is a single emission value. LDAR-Sim randomly selects one of these values to generate a new emission.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2715]: For both sample and dist: each following row right?

**Review by ThomasGalesloot — CHANGES_REQUESTED** [review]
Great work - Feel free to merge after handling this round of feedback

**Review by ThomasGalesloot — APPROVED** [review]

File: USER_MANUAL.md [hunk 26]
```diff
@@ -1549,58 +1740,981 @@ The following figure offers a visual guideline for the various parameters that c
...
+
+**Note of caution:** The `source` column is often an identical value to the `emissions_rate_source` column; however, these two columns serve different purposes. The `source` column allow users to apply the same [emission_rate_sources](#header) for different `sources`. Additionally, the `source` column, in conjunction with the `component` column, forms a compound key. In simple terms, each component/source pair must be unique and exist only once.
+
+#### repairable (Source File)
+
+**Description:** A True/False column that defines if the given emission source produces repairable(`TRUE`) or non-repairable(`FALSE`) emission.
+
+#### persistent (Source File)
+
+**Description:** A True/False column that defines if the given emission source produces persistent(`TRUE`) or non-persistent(`FALSE`) emission. Non-persistent emissions have a total active duration of the user set parameter [duration](#duration-propagating-parameter). However each active and inactive period is defined by the following parameters respectively, [active_duration](#active_duration-source-file) and [inactive_duration](#inactive_duration-source-file).
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:2496]: Can we specify that non-persistent emissions start from their emitting state?

Comment (ThomasGalesloot) [reply USER_MANUAL.md:2496]: and then cycle between emitting and not emitting

**Pull request merged**